guides/20240819_history_embedding.md

```diff
@@ -0,0 +1,237 @@
+2. **Export to CSV**: Use the extension to export your browser history as a CSV file. Save it to a convenient location, like `~/Downloads/history.csv`.
+
+Example CSV snippet:
+
+```csv
+visit_time,url,title,visit_count,typed_count
+2023-08-10 12:34:56,http://example.com,Example Site,5,2
+2023-08-09 11:22:33,http://anotherexample.com,Another Example,3,1
+```
+
+## Setting Up the Development Environment
+
+### Creating a Daytona Workspace
+
+With Daytona installed, you can create a [workspace](/definitions/d/daytona-workspace) for the [`nkkko/history` repository](https://github.com/nkkko/history):
+
+```bash
+daytona create https://github.com/nkkko/history --code
+```
+
+This command clones the [repository](/definitions/r/repository) and sets up the necessary environment. Daytona manages dependencies and configurations, ensuring that your environment is set up correctly for running the project scripts.
+
+### Set Up Environment Variables (Optional for Azure)
+
+If you plan to use Azure's [embedding](/definitions/d/embedding) services instead of the local model, you need to configure the environment variables. Create a `.env` file in the root directory of the [repository](/definitions/r/repository) and add the following:
+
+```env
+AZURE_API_VERSION=<your_azure_api_version>
+AZURE_ENDPOINT=<your_azure_endpoint>
+AZURE_OPENAI_API_KEY=<your_azure_api_key>
+```
+
+**Example**:
+```env
+AZURE_API_VERSION=2021-06-01-preview
+AZURE_ENDPOINT=https://your-resource-name.openai.azure.com/
+AZURE_OPENAI_API_KEY=your-secret-key
+```
+
+**Why Azure?**  
+Using Azure's [embedding](/definitions/d/embedding) service can be beneficial if you need higher performance or better semantic accuracy in your [embedding](/definitions/d/embedding)s, especially when dealing with large datasets.
+
+## Embedding and Searching Your Browser History
+
+### Embedding
+
+With your [development environment](/definitions/d/development-environment) ready, you can now embed your browser history into ChromaDB.
+
+To embed using the local model (all-MiniLM-L6-v2), run:
+
+```bash
+python search.py --embed path/to/your/history.csv
+```
+
+**Practical Example**:
+If your history CSV is located at `~/Downloads/history.csv`, the command would be:
+
+```bash
+python search.py --embed ~/Downloads/history.csv
+```
+
+This command reads each entry from the CSV, processes it into an [embedding](/definitions/d/embedding), and stores it in ChromaDB. The local model `all-MiniLM-L6-v2` is lightweight and works well for most purposes.
+
+**Using Azure [Embedding](/definitions/d/embedding)s**:
+If you’re leveraging Azure for [embedding](/definitions/d/embedding)s, use the `--azure` flag:
+
+```bash
+python search.py --embed ~/Downloads/history.csv --azure
+```
+
+### Searching
+
+Once your history is embedded, you can perform searches using the `search.py` script. Here's how to use it effectively:
+
+```bash
+python search.py "search query"
+```
+
+#### Refining Searches
+
```

Comment:
It would be better to have code snippets showing how things work then just reinterpreting the readme.